hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonOutputCommitter.java

```diff
@@ -282,10 +296,12 @@ private void commitJobForPartition(JobContext context, boolean overwriteSet,
         throw new IOException(e);
       }
     }
-    String segmentFileName = SegmentFileStore.genSegmentFileName(
-        loadModel.getSegmentId(), String.valueOf(loadModel.getFactTimeStamp()));
     newMetaEntry.setSegmentFile(segmentFileName + CarbonTablePath.SEGMENT_EXT);
-    newMetaEntry.setIndexSize("" + loadModel.getMetrics().getMergeIndexSize());
+    if (isMergeIndex) {
```

Comment:
can you move it in the else case of same method line 280, scattered checks are hard to read. It is better to keep together

Reply:
loadModel.getMetrics().getMergeIndexSize() filled in MergeIndexEventListene.onEvent() when mergeindex is created. So, can't make it else case to line 280.